configure/CMakeLists.txt

```diff
@@ -214,7 +214,7 @@ if(NOT WIN32)
             else()
                 add_compile_options(-O0)
             endif()
-	    add_compile_options(-Wall -Wextra -pedantic)
+            add_compile_options(-Wall -Wextra -pedantic -fstack-protector-strong -Wformat -Werror=format-security)
```

Comment:
Isn't stack-protector something we may consider enabling in release builds, not in debug builds? Per my understanding it is a security feature against attacks based on stack corruption, not a debugging facility. For debugging purposes an address sanitizer job in CI may be another option.

Reply:
Good point. I've changed that. I also incorporated the ideas from https://github.com/tango-controls/cppTango/pull/662#discussion_r419463681.